app/gui/view/execution-environment-selector/src/lib.rs

```diff
@@ -244,23 +251,20 @@ impl component::Frp<Model> for Frp {
 
             // == Inputs ==
 
-            eval input.set_available_execution_modes ((entries) model.set_entries(entries.clone()));
+            eval input.set_available_execution_environments ((entries) model.set_entries(entries.clone()));
+
+            update_selected_entry <- input.set_execution_environment.map2(&input.set_available_execution_environments, |entry, entries| {
+                    entries.iter().position(|mode| mode == entry)
+            });
+            dropdown.frp.set_selected <+ update_selected_entry;
 
             selected_id <- dropdown.frp.chosen_entry.unwrap();
-            selection <- all(input.set_available_execution_modes, selected_id);
+            selection <- all(input.set_available_execution_environments, selected_id);
             selected_entry <- selection.map(|(entries, entry_id)| entries[*entry_id].clone());
-            output.selected_execution_mode <+ selected_entry;
+            output.selected_execution_environment <+ selected_entry;
 
             eval selected_entry ([model] (execution_mode) {
-                // TODO(#5930): Revisit when connecting with externally set execution mode
-                let play_button_visibility = match execution_mode.as_str() {
-                    "design" => true,
-                    "live" => false,
-                    _ => {
-                        error!("Play button: invalid execution mode");
-                        false
-                    }
-                };
+                let play_button_visibility = matches!(execution_mode.to_lowercase().as_str(), "design");
```

Comment:
Why `TODO` was removed? The logic hasn't changed, has it?

Reply:
This is implementing the referenced issue (5930) and it appears there is nothing else to do, so no point in keeping the reference to a son to be closed task. This probably should be revisited when we get the custom modes that Wojciech mentioned din one of his other comments. 

I guess if we really want to be prepared for it, we could already an API that allows us to set which execution environments should show the play button, but for the current spec that is not needed.